### ISSUE TITLE
Wilder completion window overshadowed, fuzzy autocomplete also not working within wilder

### ISSUE THREAD

**Issue by mihneaCR** [jdhao/nvim-config#47]
- `nvim --version`: 0.6.1
- terminal emulator and version: Neovide/Alacritty
- Operating system/version: Archlinux 5.17.1-arch1-1
- Have you read and followed all steps in the [doc](https://github.com/jdhao/nvim-config/tree/master/docs): yes
- Output of command `set runtimepath`:

```
runtimepath=~/.config/nvim,/etc/xdg/nvim,~/.local/share/nvim/site,
~/.local/share/nvim/site/pack/packer/opt/vim-indent-object,
~/.local/share/nvim/site/pack/packer/opt/whitespace.nvim,
~/.local/share/nvim/site/pack/packer/opt/vim-commentary,
~/.local/share/nvim/site/pack/packer/opt/which-key.nvim,
~/.local/share/nvim/site/pack/packer/opt/nvim-lspconfig,
~/.local/share/nvim/site/pack/packer/opt/cmp-nvim-lsp,
~/.local/share/nvim/site/pack/packer/opt/cmp-omni,
~/.local/share/nvim/site/pack/packer/opt/cmp-path,
~/.local/share/nvim/site/pack/packer/opt/cmp-nvim-lua,
~/.local/share/nvim/site/pack/packer/opt/cmp-buffer,
~/.local/share/nvim/site/pack/packer/opt/nvim-cmp,
~/.local/share/nvim/site/pack/packer/opt/lspkind-nvim,
~/.local/share/nvim/site/pack/packer/opt/indent-blankline.nvim,
~/.local/share/nvim/site/pack/packer/opt/vim-asterisk,
~/.local/share/nvim/site/pack/packer/opt/unicode.vim,
~/.local/share/nvim/site/pack/packer/opt/vim-swap,
~/.local/share/nvim/site/pack/packer/opt/hop.nvim,
~/.local/share/nvim/site/pack/packer/opt/lualine.nvim,
~/.local/share/nvim/site/pack/packer/opt/targets.vim,
~/.local/share/nvim/site/pack/packer/opt/vim-sandwich,
~/.local/share/nvim/site/pack/packer/opt/vim-repeat,
~/.local/share/nvim/site/pack/packer/opt/bufferline.nvim,
~/.local/share/nvim/site/pack/packer/opt/alpha-nvim,
~/.local/share/nvim/site/pack/packer/opt/nvim-web-devicons,
~/.local/share/nvim/site/pack/packer/opt/vim-matchup,
~/.local/share/nvim/site/pack/packer/opt/vim-signify,
~/.local/share/nvim/site/pack/packer/opt/nvim-treesitter,
~/.local/share/nvim/site/pack/packer/opt/nvim-notify,
~/.local/share/nvim/site/pack/*/start/*,
~/.local/share/nvim/site/pack/packer/opt/sonokai,
~/.local/share/nvim/site/pack/packer/opt/nvim-gdb,
~/.local/share/nvim/site/pack/packer/opt/committia.vim,
~/.local/share/nvim/site/pack/packer/opt/wilder.nvim,
~/.local/share/nvim/site/pack/packer/opt/packer.nvim,/usr/local/share/nvim/site,
/usr/share/nvim/site,/usr/share/nvim/runtime,/usr/lib/nvim,
~/.local/share/nvim/site/pack/packer/opt/cmp-nvim-lsp/after,
~/.local/share/nvim/site/pack/packer/opt/cmp-omni/after,
~/.local/share/nvim/site/pack/packer/opt/cmp-path/after,
~/.local/share/nvim/site/pack/packer/opt/cmp-nvim-lua/after,
~/.local/share/nvim/site/pack/packer/opt/cmp-buffer/after,
~/.local/share/nvim/site/pack/packer/opt/vim-sandwich/after,
~/.local/share/nvim/site/pack/packer/opt/vim-matchup/after,
~/.local/share/nvim/site/pack/packer/opt/nvim-treesitter/after,
~/.local/share/nvim/site/pack/packer/opt/sonokai/after,
/usr/share/nvim/site/after,/usr/local/share/nvim/site/after,
~/.local/share/nvim/site/after,/etc/xdg/nvim/after,~/.config/nvim/after,
/usr/share/vim/vimfiles
```

**Describe the bug**
Autocomplete in Command Line opens up other "completion" next to Wilder.nvim, does also not include fuzzy matching (e.g. expanding VimComp to VimtexCompile) 

Steps to reproduce the issue (Fuzzy completion):
1.Open neovim
2.Press :, type some command(halfway) (for example VimComp, used to autocomplete to VimtexCompile)
3.Press Tab to autocomplete

**Expected behavior** (Fuzzy Completion)
Should autocomplete command, instead nothing happens. (This is for fuzzy completion)

Steps to reproduce the issue (Normal completion):
1.Open neovim
2.Press :, type some command(halfway) (for example :e l, if there are multiple files starting with l)
3.Press Tab to autocomplete

**Expected behavior** (Normal Completion)
Should open wilder, with the "cursor" set to the first option, instead opens some other autocomplete utility. (see attached screenshot, if you look closely you can still see the wilder popup below)

![before-tab](https://user-images.githubusercontent.com/47532611/162580238-508ba963-b1bd-4a0c-9d7c-18b0f3a5d35a.jpeg)

![after-tab](https://user-images.githubusercontent.com/47532611/162580251-eb345663-74ec-4fe6-8103-6358ff968abd.jpeg)

**Comment by mihneaCR** [comment]
> * `nvim --version`: 0.6.1
> 
>     * terminal emulator and version: Neovide/Alacritty
> 
>     * Operating system/version: Archlinux 5.17.1-arch1-1
> 
>     * Have you read and followed all steps in the [doc](https://github.com/jdhao/nvim-config/tree/master/docs): yes
> 
>     * Output of command `set runtimepath`:runtimepath=~/.config/nvim,/etc/xdg/nvim,~/.loca
>       l/share/nvim/site,~/.local/share/nvim/site/pack/pa
>       cker/opt/vim-indent-object,~/.local/share/nvim/sit
>       e/pack/packer/opt/whitespace.nvim,~/.local/share/n
>       vim/site/pack/packer/opt/vim-commentary,~/.local/s
>       hare/nvim/site/pack/packer/opt/which-key.nvim,~/.l
>       ocal/share/nvim/site/pack/packer/opt/nvim-lspconfi
>       g,~/.local/share/nvim/site/pack/packer/opt/cmp-nvi
>       m-lsp,~/.local/share/nvim/site/pack/packer/opt/cmp
>       -omni,~/.local/share/nvim/site/pack/packer/opt/cmp
>       -path,~/.local/share/nvim/site/pack/packer/opt/cmp
>       -nvim-lua,~/.local/share/nvim/site/pack/packer/opt
>       /cmp-buffer,~/.local/share/nvim/site/pack/packer/o
>       pt/nvim-cmp,~/.local/share/nvim/site/pack/packer/o
>       pt/lspkind-nvim,~/.local/share/nvim/site/pack/pack
>       er/opt/indent-blankline.nvim,~/.local/share/nvim/s
>       ite/pack/packer/opt/vim-asterisk,~/.local/share/nv
>       im/site/pack/packer/opt/unicode.vim,~/.local/share
>       /nvim/site/pack/packer/opt/vim-swap,~/.local/share
>       /nvim/site/pack/packer/opt/hop.nvim,~/.local/share
>       /nvim/site/pack/packer/opt/lualine.nvim,~/.local/s
>       hare/nvim/site/pack/packer/opt/targets.vim,~/.loca
>       l/share/nvim/site/pack/packer/opt/vim-sandwich,~/.
>       local/share/nvim/site/pack/packer/opt/vim-repeat,~
>       /.local/share/nvim/site/pack/packer/opt/bufferline
>       .nvim,~/.local/share/nvim/site/pack/packer/opt/alp
>       ha-nvim,~/.local/share/nvim/site/pack/packer/opt/n
>       vim-web-devicons,~/.local/share/nvim/site/pack/pac
>       ker/opt/vim-matchup,~/.local/share/nvim/site/pack/
>       packer/opt/vim-signify,~/.local/share/nvim/site/pa
>       ck/packer/opt/nvim-treesitter,~/.local/share/nvim/
>       site/pack/packer/opt/nvim-notify,~/.local/share/nv
>       im/site/pack/_/start/_,~/.local/share/nvim/site/pa
>       ck/packer/opt/sonokai,~/.local/share/nvim/site/pac
>       k/packer/opt/nvim-gdb,~/.local/share/nvim/site/pac
>       k/packer/opt/committia.vim,~/.local/share/nvim/sit
>       e/pack/packer/opt/wilder.nvim,~/.local/share/nvim/
>       site/pack/packer/opt/packer.nvim,/usr/local/share/
>       nvim/site,/usr/share/nvim/site,/usr/share/nvim/run
>       time,/usr/lib/nvim,~/.local/share/nvim/site/pack/p
>       acker/opt/cmp-nvim-lsp/after,~/.local/share/nvim/s
>       ite/pack/packer/opt/cmp-omni/after,~/.local/share/
>       nvim/site/pack/packer/opt/cmp-path/after,~/.local/
>       share/nvim/site/pack/packer/opt/cmp-nvim-lua/after
>       ,~/.local/share/nvim/site/pack/packer/opt/cmp-buff
>       er/after,~/.local/share/nvim/site/pack/packer/opt/
>       vim-sandwich/after,~/.local/share/nvim/site/pack/p
>       acker/opt/vim-matchup/after,~/.local/share/nvim/si
>       te/pack/packer/opt/nvim-treesitter/after,~/.local/
>       share/nvim/site/pack/packer/opt/sonokai/after,/usr
>       /share/nvim/site/after,/usr/local/share/nvim/site/
>       after,~/.local/share/nvim/site/after,/etc/xdg/nvim
>       /after,~/.config/nvim/after,/usr/share/vim/vimfile
>       s
> 
> 
> **Describe the bug** Autocomplete in Command Line opens up other "completion" next to Wilder.nvim, does also not include fuzzy matching (e.g. expanding VimComp to VimtexCompile)
> 
> Steps to reproduce the issue (Fuzzy completion): 1.Open neovim 2.Press :, type some command(halfway) (for example VimComp, used to autocomplete to VimtexCompile) 3.Press Tab to autocomplete
> 
> **Expected behavior** (Fuzzy Completion) Should autocomplete command, instead nothing happens. (This is for fuzzy completion)
> 
> Steps to reproduce the issue (Normal completion): 1.Open neovim 2.Press :, type some command(halfway) (for example :e l, if there are multiple files starting with l) 3.Press Tab to autocomplete
> 
> **Expected behavior** (Normal Completion) Should open wilder, with the "cursor" set to the first option, instead opens some other autocomplete utility. (see attached screenshot, if you look closely you can still see the wilder popup below)
> 
> ![before-tab](https://user-images.githubusercontent.com/47532611/162580238-508ba963-b1bd-4a0c-9d7c-18b0f3a5d35a.jpeg)
> 
> ![after-tab](https://user-images.githubusercontent.com/47532611/162580251-eb345663-74ec-4fe6-8103-6358ff968abd.jpeg)

Added:Seems to be an issue with Wilder, reverting to commit addb378c2ae41d3d4fbd214ca525eb1f2b2855e5 while keeping the same configuration solves the issue.

**Comment by jdhao** [comment]
>  expanding VimComp to VimtexCompile

It is not clear whether autocompletion for command does not work at all or just does not work for this case. `VimtexCompile` is provided by VimTex, which is only activated for tex files (see [here](https://github.com/jdhao/nvim-config/blob/639b3d60671ed5c71c66bc9c36132d31cb3a85f1/lua/plugins.lua#L282)). Please make sure you are editting a tex file.

> Should open wilder, with the "cursor" set to the first option, instead opens some other autocomplete utility. (see attached screenshot, if you look closely you can still see the wilder popup below)

I am not what do you mean. Please make it clear. All I can see from the screenshot is that wilder.nvim is working and shows the completion for relevant files and folders.

**Comment by jdhao** [comment]
Also regarding VimTex, I only enable VimTex for Windows and macOS, which is clearly stated in the README.

**Comment by mihneaCR** [comment]
> > expanding VimComp to VimtexCompile
> 
> It is not clear whether autocompletion for command does not work at all or just does not work for this case. `VimtexCompile` is provided by VimTex, which is only activated for tex files (see [here](https://github.com/jdhao/nvim-config/blob/639b3d60671ed5c71c66bc9c36132d31cb3a85f1/lua/plugins.lua#L282)). Please make sure you are editting a tex file.
I was editing a tex file. Autocomplete works as intended for code (with fuzzy matching also), however it does not work for autocompletion of commands/filepaths, as the window in the second screenshot is overlayed over wilder, and accepts commands instead)
If trying to expand fuzzy commands, nothing happens.
> 
> > Should open wilder, with the "cursor" set to the first option, instead opens some other autocomplete utility. (see attached screenshot, if you look closely you can still see the wilder popup below)
> 
> I am not what do you mean. Please make it clear. All I can see from the screenshot is that wilder.nvim is working and shows the completion for relevant files and folders.
I in the screenshot, when pressing tab after typing :e l, it should focus lua/, instead, the window in the second screenshot opens over wilder (which you can faintly see in the background of the second picture).

> Also regarding VimTex, I only enable VimTex for Windows and macOS, which is clearly
> stated in the README. If you need to use VimTex on Linux, you need to tweak the code a > bit.
I did use it on linux without problem, this is not a problem with Vimtex, i just used Vimtex to illustrate the behavior of the fuzzy completion that does not work anymore.
VimtexCompile was suggested as an option (meaning that vimtex is loaded), however it did not expand to VimtexCompile.

However, i mangaged to fix this issue by downgrading wilder to commit addb378c2ae41d3d4fbd214ca525eb1f2b2855e5. This seems to be an incompatibility between the config and the newest commit.

**Comment by jdhao** [comment]
Please use proper markdown, the current comment is hard to read, 😿

**Comment by jdhao** [comment]
I tried the latest version of Vimtex and it works perfectly for me. 

<img width="337" alt="image" src="https://user-images.githubusercontent.com/16662357/162581556-d9cfd83b-bf24-4565-b0fa-6f502f015b3f.png">

Maybe it is an issue with your settings or environment.

**Comment by mihneaCR** [comment]
> Please use proper markdown, the current comment is hard to read, crying_cat_face
I'm sorry, I'm new to github.



> I tried the latest version of Vimtex and it works perfectly for me.
> 
> <img alt="image" width="337" src="https://user-images.githubusercontent.com/16662357/162581556-d9cfd83b-bf24-4565-b0fa-6f502f015b3f.png">
> 
> Maybe it is an issue with your settings or environment.
It is not a problem with Vimtex, but with wilder.nvim. Vimtex works as intended. Try updating Wilder.nvim, type  :vimcomp, and press tab. It does not jump to VimtexCompile which you can then select.
Or try completing anything without a fuzzy matching.
Anyway, you can close this thread, everything works now!

**Comment by jdhao** [comment]
Okay, I see the issue. So the issue is you can not use `Tab` to navigate the wilder completion window, right? I can reproduce the issue. I think it has something to do with the latest update of wilder.nvim.

**Comment by mihneaCR** [comment]
> Okay, I see the issue. So the issue is you can not use `Tab` to navigate the wilder completion window, right? I can reproduce the issue. I think it has something to do with the latest update of wilder.nvim.

Yes, exactly. However, i tried downgrading, and this unfortunately doesn't help, i found that this behavior is specific to some folders.
It does not happen in the .config/nvim folder, but in other folders it does, even with older versions of wilder.nvim (tried b59648ad8588bcba377f4eecdea317796ebd1f9d) also. Strangely, everything used to work perfectly this morning.

**Comment by jdhao** [comment]
I will keep an eye on this issue to see if I can find the cause.

**Comment by mihneaCR** [comment]
> I will keep an eye on this issue to see if I can find the cause.

Thank you!
Sorry again for the confusing description, I am still new to all of this.

**Comment by jdhao** [comment]
@mihneaCR This issue has been fixed in the latest commit: https://github.com/jdhao/nvim-config/commit/3134f088cd2ad870474b0d68c7242b4473aab7f9. Please update if you are interested.

Thanks for your report!

**Comment by mihneaCR** [comment]
> @mihneaCR This issue has been fixed in the latest commit: [3134f08](https://github.com/jdhao/nvim-config/commit/3134f088cd2ad870474b0d68c7242b4473aab7f9). Please update if you are interested.
> 
> Thanks for your report!

Thanks for helping, everything works fine now!
